Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -1411,3 +1411,70 @@ lemma SuccOrder.forall_ne_bot_iff
   rw [← Nat.succ_pred_eq_of_pos hj]
   simp only [Function.iterate_succ', Function.comp_apply]
   apply h
+
+section OrdConnected
+
+variable {α : Type*} [PartialOrder α]
+
+open scoped Classical in
+noncomputable instance Set.OrdConnected.predOrder [PredOrder α] (s : Set α) [s.OrdConnected] :
```

Comment:
Does this make sense as an instance? Surely it's a bit weird that `pred x` for `x ∈ s` could mean either of `pred ↑x` or `pred x` depending on whether this fires or not.

Reply:
I don't think it's problematic, writing `pred x` just gives `failed to synthesize PredOrder ↑s` without this triggering.